packages/next/src/client/components/router-reducer/create-router-cache-key.ts

```diff
@@ -5,9 +5,17 @@ export function createRouterCacheKey(
   segment: Segment,
   withoutSearchParameters: boolean = false
 ) {
-  return Array.isArray(segment)
-    ? `${segment[0]}|${segment[1]}|${segment[2]}`.toLowerCase()
-    : withoutSearchParameters && segment.startsWith(PAGE_SEGMENT_KEY)
-    ? PAGE_SEGMENT_KEY
-    : segment
+  // if the segment is an array, it means it's a dynamic segment
```

Comment:
No change in behavior to this file, just split it out of a ternary to better annotate with comments